tbot_contrib/time_generic_testcases/time_tescases.py

```diff
@@ -0,0 +1,101 @@
+from termcolor import colored
+import statistics
+import time
+import tbot
+import typing
+
+
+@tbot.testcase
+def testcase_clockTime(
+    testcase: typing.Callable, *args: typing.Any, **kwargs: typing.Any
+) -> float:
+    """Return the time that the given testcase took
+    you can pass any arguments that the testcase needs"""
+    start = time.time()
+    testcase(*args, **kwargs)
+    finish = time.time()
+    return finish - start
+
+
+@tbot.testcase
+def statistics_clockTime(
+    testcase: typing.Callable, *args: typing.Any, **kwargs: typing.Any
+) -> None:
+    """receive a testcase and prints descriptive statics
+    you can pass as an argument:
+    times = the number of times that the testcase is to be repeated
+    sleep = the time in seconds that the bucle must wait
+
+    you can pass any arguments that the testcase needs """
+
+    if "times" in kwargs:
+        times = kwargs.pop("times")
+    else:
+        times = 100
+
+    if "sleep" in kwargs:
+        sleep = kwargs.pop("sleep")
+    else:
+        sleep = 0
+
+    arrayElapsedTimes = []
+
+    for n in range(times):
+
+        elapesedTime = testcase_clockTime(testcase, *args, **kwargs)
+        arrayElapsedTimes.append(elapesedTime)
+        time.sleep(sleep)
+
+    mean = statistics.mean(arrayElapsedTimes)
+    harmonicMean = statistics.harmonic_mean(arrayElapsedTimes)
+    median = statistics.median(arrayElapsedTimes)
+    variance = statistics.pvariance(arrayElapsedTimes)
+    standardDeviation = statistics.pstdev(arrayElapsedTimes)
+    coloredText = (
+        colored("mean:", "green")
+        + " {0}"
+        + colored("  harmonic_mean:", "green")
+        + " {1}"
+        + colored("  median:", "green")
+        + " {2}"
+        + colored("  variance:", "green")
+        + " {3}"
+        + colored("  starndard_deviation:", "green")
+        + " {4}"
+    )
+    tbot.log.message(
+        coloredText.format(mean, harmonicMean, median, variance, standardDeviation)
+    )
```

Comment:
In tbot I generally use [`tbot.log.c()`](https://tbot.tools/modules/log.html#tbot.log.c) for coloring and [f-Strings](https://cito.github.io/blog/f-strings/) for formatting.  So this would be:

```python
tbot.log.message(
    f"""\
Timing Results:
    {tbot.log.c('mean').green}: {results.mean}
    {tbot.log.c('harmonic mean').green}: {results.harmonic_mean}
    {tbot.log.c('median').green}: {results.median}
    {tbot.log.c('variance').green}: {results.pvariance}
    {tbot.log.c('standard deviation').green}: {results.pstdev}
"""
)
```